### PR TITLE
Makes pod people a phytosian subtype

### PR DESCRIPTION
**Pull request by Time-Green**
Bees wont attack phytosians anymore. Wich was moved from pod to phytosian.
Also adds check for both the species, and one for species with the PLANT flag.

Vaults now spawn Phytosians instead of pod people. They can be both gender and pull from a txt with names.

Also some other phytosian/pod constistency.

Roundstart plant names is something for a future PR, for the time being, the txt file with the surnames is just there.

:cl:
tweak: Makes podmen a phytosian subtype
/:cl:

### CODE REVIEW

**Comment by Time-Green** [comment]
My branch got extremely fucked up, ignore that. Ill fix it later.

**Comment by ghost** [comment]
stealth buff to phytosians

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -29,6 +29,9 @@
 #define isashwalker(A) (is_species(A, /datum/species/lizard/ashwalker))
 #define ischiefwalker(A) (is_species(A, /datum/species/lizard/ashwalker/chieftain))
 #define iskrampus(A) (is_species(A, /datum/species/demon))
+#define ispod(A) (is_species(A, /datum/species/pod))
+#define isphytosian(A) (is_species(A, /datum/species/plant))
```
Comment (ShadowDeath6) [inline code/__DEFINES/is_helpers.dm]: gotta correct these species references

File: config/names/last_plant.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Palm
+Pine
+Poplar
+Redwood
+Spruce
+Sycamore
+Willow
+Yew
+Vine
+Dandylion
```
Comment (ShadowDeath6) [inline config/names/last_plant.txt]: dandelion

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 3]
```diff
@@ -801,62 +801,11 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
...
-				if(prob(80))
-					randmutb(H)
-				else
-					randmutg(H)
-				H.domutcheck()
-			else
-				H.adjustFireLoss(rand(5,15))
-				H.show_message("<span class='userdanger'>The radiation beam singes you!</span>")
-		if(/obj/item/projectile/energy/florayield)
-			H.nutrition = min(H.nutrition+30, NUTRITION_LEVEL_FULL)
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/species_types.dm]: where is all this moved

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 4]
```diff
@@ -465,7 +465,7 @@ datum/species/lizard/before_equip_job(datum/job/J, mob/living/carbon/human/H)
 	name = "Phytosian"
 	id = "plant"
 	default_color = "59CE00"
-	specflags = list(MUTCOLORS,EYECOLOR,THRALLAPPTITUDE)
+	specflags = list(MUTCOLORS,EYECOLOR,THRALLAPPTITUDE, PLANT)
 	attack_verb = "slice"
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/human/species_types.dm]: this isn't in the scope of the pr but this ends up being "sliceed", can you change it to "slic"

File: code/modules/projectiles/projectile/special.dm [hunk 5]
```diff
@@ -93,15 +93,6 @@
...
 
-/obj/item/projectile/energy/floramut/on_hit(atom/target, blocked = 0)
-	. = ..()
-	if(iscarbon(target))
-		var/mob/living/carbon/C = target
-		if(C.dna.species.id == "pod")
-			randmuti(C)
-			randmut(C)
-			C.updateappearance()
-			C.domutcheck()
```
Comment (ShadowDeath6) [inline code/modules/projectiles/projectile/special.dm:104]: is this moved to phyto hit reactions

Comment (Time-Green) [reply code/modules/projectiles/projectile/special.dm:104]: Phytosians already had it. If I were to leave it, they'd get double mutations

File: code/__DEFINES/is_helpers.dm [hunk 6]
```diff
@@ -29,6 +29,9 @@
 #define isashwalker(A) (is_species(A, /datum/species/lizard/ashwalker))
 #define ischiefwalker(A) (is_species(A, /datum/species/lizard/ashwalker/chieftain))
 #define iskrampus(A) (is_species(A, /datum/species/demon))
+#define ispod(A) (is_species(A, /datum/species/plant/pod))
+#define isphytosian(A) (is_species(A, /datum/species/plant))
+#define isplant(A) (PLANT in A.dna.species.specflags & PLANT)
```
Comment (Cruix) [inline code/__DEFINES/is_helpers.dm]: remove the & plant

**Comment by Cruix** [comment]
Why not just make pod people still have their plant-based bonuses over the special snowflake roundstart plant race so botanists have to kill themselves to be a special snowflake botanist like they did in the old days?

**Comment by Time-Green** [comment]
Most of their code was just outdated phytosian code, that's why.
If you think it shouldn't, just close this, it's a small PR

**Pull request merged**